xapian-applications/omega/postprocess.py

```diff
@@ -161,12 +149,36 @@ def generate_query_file(final_log, query_file):
     098f6bcd4621d373cade4e832627b4f6,test
     """
     with open(final_log, 'r') as s, open(query_file, 'w+') as w:
-        reader = csv.DictReader(s, delimiter=',')
+        reader = csv.DictReader(s)
         writer = csv.writer(w)
 
         for row in reader:
             writer.writerow([row['QueryID'], row['Query']])
 
+
+def test():
+    pass
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='''Postprocess click data files.
+
+        This script generates the final clickstream log file from input search and
+        click log files and creates Query file that can be used by Xapian Letor
+        module for generating its training files.
+        ''', formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("search_log", type=str, help="Path to the search.log file.")
+    parser.add_argument("clicks_log", type=str, help="Path to the clicks.log file.")
+    parser.add_argument("final_log", type=str, help="Path to save final.log file.")
+    parser.add_argument("query_file", type=str, help="Path to save query.txt file.")
+    parser.add_argument("--test", type=test, action='store_true', help="Run tests for this script.")
+    args = parser.parse_args()
+
+if args.test:
+    test()
+
```

Comment:
Strictly, there are too many newlines here, although for some reason flake8 doesn't pick that up. (I suspect it may be because of the main guards.)